### PR TITLE
date32 column save to Jay + read from Jay

### PR DESCRIPTION
**Pull request by st-pasha**
Implement serializing date32 columns into Jay, and reading the file back.

Note: I had to edit `jay_generated.h` manually  inserting 
```
if (!obj) return true;
```
at line 611, because otherwise old Jay files were reported as being invalid.

The `flatbuffers` library used in our project updated as well to the most recent version (it hasn't been updated since 2018).

WIP for #2858

### CODE REVIEW

**Comment by oleksiyskononenko** [comment]
Seems like `tests/types/test-date32.py` was updated from some other PR, should we should merge `main` first.

File: src/core/jay/jay_generated.h [hunk 1]
```diff
@@ -23,8 +23,10 @@ struct StatsFloat32;
 struct StatsFloat64;
 
 struct Frame;
+struct FrameBuilder;
```
Comment (oleksiyskononenko) [inline src/core/jay/jay_generated.h:26]: How this `jay_generated.h` file is generated?

File: tests/types/test-date32.py [hunk 2]
```diff
@@ -127,6 +127,23 @@ def test_from_numpy_with_nats(np):
...
+def test_save_to_jay(tempfile_jay):
+    d = datetime.date
+    src = [d(1, 1, 1), d(2001, 12, 13), d(2026, 5, 9), None, d(1956, 11, 11)]
+    DT = dt.Frame(src)
+    DT.to_jay(tempfile_jay)
+    del DT
+    DT2 = dt.fread(tempfile_jay)
+    assert DT2.shape == (5, 1)
+    assert DT2.types == [dt.Type.date32]
+    assert DT2.to_list()[0] == src
```
Comment (oleksiyskononenko) [inline tests/types/test-date32.py:143]: Do we also want to calculate stats on the date column and save/read it to/from jay?

**Review by oleksiyskononenko — APPROVED** [review]

File: src/core/jay/jay_generated.h (hunk 1 above)
Comment (st-pasha) [reply src/core/jay/jay_generated.h:26]: There is a comment about this in `jay.fbs`: you run `flatc --cpp jay.fbs`, where `flatc` must be installed through either `brew` or `apt-get` or whatever mechanism is available on your system.

File: tests/types/test-date32.py (hunk 2 above)
Comment (st-pasha) [reply tests/types/test-date32.py:143]: ok

**Pull request merged**